ClinicalGenomicsGBG/qd-rnaseq-wrapper
# Initiate runner

# Initiate runner via wrapper
This PR adds quite a bit of new functionality, mainly that the wrapper now queries SLIMS for any new samples marked with "QD-RNA" and runns the nf-core/rnaseq and nf-core/rnafusion on all such samples.
In order to not run a sample multiple times, a text file with previously analysed samples is kept, much like how demuxer or WOPR keeps tracks. 

The intention is to remove this file and have SLIMS keeping track of which samples have been already analysed, and the wrapper should query this.

### New stuff
- Queries SLIMS for any samples marked as "QD-RNA". These functions were largely copied from WOPR. This should really be a separate slims API in the overall CGG repo. Done like this for now.
- Now have a passwords file copied from WOPR to keep credentials for SLIMS
- Generates the input samplesheets for the nf-core pipelines based on SLIMS information regarding where fastq files are
- Starts the runner in a separate thread for all samples, meaning that quite a lot of instances of nextflow could be started
- Improved logging in the wrapper log. If there are no new samples, nothing gets written to the log. This was done in order to keep it from being spammed.
- Click CLI was updated to have the runner be both used with the CLI as well as imported into the wrapper

### To test
To test this, you can try to just set the secondary analysis of RNA sample (or you could use a DNA sample, why not) in SLIMS to "QD-RNA". The wrapper should then find all of them.
```
git clone git@github.com:ClinicalGenomicsGBG/qd-rnaseq-wrapper.git
cd qd-rnaseq-wrapper
git checkout initiate_runner
#(optional step if no conda env built)
conda env create -f conda-env/qd-rnaseq.yaml -p conda-env/qd-rnaseq
conda activate conda-env/qd-rnaseq
python wrapper.py
```
